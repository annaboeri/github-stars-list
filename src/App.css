/* Color Palette */
:root {
  --gray-dark: #575A6C;
  --gray-light: #f1f1f1;
  --green-faint: #e0e2d21c;
  --green-light: #E0E2D2;
  --green-medium: #B4C540;
  --blue-medium: #3686C9;
}

/* Global Styles */
ul {
  list-style-type: none;
  margin-block-start: 0;
  margin-block-end: 0;
  padding-inline-start: 0;
}

h1 {
  color: white;
  font-size: calc(21px + 4vmin);
  margin: 0.5em;
}

h2 {
  color: var(--green-light);
  font-size: 1.25em;
}

h3 {
  color: var(--blue-medium);
  font-size: 1.25em;
}

p, button, i {
  color: var(--gray-dark);
}

a, a:visited {
  color: var(--gray-dark);
  text-decoration: none;
}

a:hover {
  color: var(--green-medium);
}

img {
  max-width: 100%;
}

button {
  background-color: var(--green-light);
  border: 1px solid var(--gray-light);
  border-radius: 3px;
  font-weight: 600;
  margin: 10px;
  padding: 8px 10px;
}

button:hover, button:active {
  background-color: var(--green-medium);
}

button:focus {
  outline: none;
}

.app {
  text-align: center;
}

.ellipsis {
  display: -webkit-box;
  line-height: 20px;     /* IE Safari fallback */
  max-height: 60px;       /* IE Safari fallback */
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

@media screen and (max-width: 40em) {
  h2 {
    font-size: 1em;
  }
}

/* Global Icon Styles */
.fas, .fab, .far {
  padding-left: 15px;
  padding-right: 5px;
}

.fa-spinner {
  padding: 0;
  margin: 10px;
}
