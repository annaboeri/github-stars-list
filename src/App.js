import React from 'react';
import Header from './header/Header.js';
import Footer from './footer/Footer.js';
import RepoListContainer from './repo-list/RepoListContainer.js';
import './App.css';

const App = () => {
  return (
    <div className="app">
      <Header />
      <RepoListContainer />
      <Footer />
    </div>
  );
}

export default App;
