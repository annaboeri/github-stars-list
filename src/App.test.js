import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import App from './App';

describe('Header Component', () => {
  it('should render app components correctly', (done) => {
    const component = shallow(<App />);
    const tree = toJson(component);

    expect(tree).toMatchSnapshot();
    expect(component.find('Header').exists()).toBe(true);
    expect(component.find('RepoListContainer').exists()).toBe(true);
    expect(component.find('Footer').exists()).toBe(true);
    done();
  });
});
