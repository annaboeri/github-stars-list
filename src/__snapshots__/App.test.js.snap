// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render app components correctly 1`] = `
<div
  className="app"
>
  <Header />
  <RepoListContainer />
  <Footer />
</div>
`;
